orion/pkg/repos/location_repo.go

```diff
@@ -161,26 +162,28 @@ func (lr *locationRepo) Insert(location domains.Location) error {
 	return handleSqlExecResult(result, 1, "location was not inserted")
 }
 
-func (lr *locationRepo) Publish(locIds []string) []domains.LocationErrorBody {
-	errors := make([]domains.LocationErrorBody, 0)
+func (lr *locationRepo) Publish(locIds []string) []domains.PublishErrorBody {
+	errorList := make([]domains.PublishErrorBody, 0)
 
 	for _, locId := range locIds {
 		location, err := lr.SelectByLocationId(locId)
 		if err != nil {
-			errors = append(errors, domains.LocationErrorBody{LocId: locId, Error: err})
+			errorList = append(errorList, domains.PublishErrorBody{StringId: locId, Error: err})
 			continue
 		}
 		if !location.PublishedAt.Valid {
 			now := time.Now().UTC()
 			location.PublishedAt.Scan(now)
 			err = lr.Update(locId, location)
 			if err != nil {
-				errors = append(errors, domains.LocationErrorBody{LocId: locId, Error: err})
+				errorList = append(errorList, domains.PublishErrorBody{StringId: locId, Error: err})
 			}
+		} else {
+			errorList = append(errorList, domains.PublishErrorBody{StringId: locId, Error: errors.New("location is already published")})
```

Comment:
I think we agreed that there should be no error.

Reply:
I think we agreed that the overall process would still go through, but I don't know if we agreed on whether the individual thing should be treated as an error.
In my opinion it makes sense to treat it as an error, since the option to publish an already published item from the user side shouldn't be available in the first place.

Reply:
We agreed on idempotency earlier.

Reply:
I don't know how such a scenario would arise but ok.